src/doom/m_random.c

```diff
@@ -63,3 +63,10 @@ void M_ClearRandom (void)
 {
     rndindex = prndindex = 0;
 }
+
+// inspired by the same routine in Eternity, thanks haleyjd
+int P_SubRandom (void)
+{
+    int r = P_Random();
```

Comment:
as long as the order of evaluation remained the same it wouldn't matter; if the order changed the optimiser was broken.